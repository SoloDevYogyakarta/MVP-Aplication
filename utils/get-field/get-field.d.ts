import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
