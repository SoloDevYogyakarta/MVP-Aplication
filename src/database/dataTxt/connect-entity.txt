{"id":895,"source_id":"k8TJ_2ROjCwho9XJlfZRl","foreign_id":"Dk1u1tKFZvPDubveGZhUc","public_id":"8kOUYHeXhm9NpFeAcczOI"}