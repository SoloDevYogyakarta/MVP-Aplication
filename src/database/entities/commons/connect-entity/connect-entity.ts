import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ConnectEntity extends BaseEntity {
  source_id: string;
  foreign_id: string;
}

export type ConnectCreationAttribute = Optional<ConnectEntity, 'id'>;
export interface ConnectInstance
  extends Model<ConnectCreationAttribute, ConnectEntity>,
    ConnectEntity {}
export const connectEntity = sequelize.define<ConnectInstance>(
  'CONNECT',
  {
    public_id: DataTypes.STRING,
    source_id: DataTypes.STRING,
    foreign_id: DataTypes.STRING,
  } as ModelAttributes<ConnectInstance>,
  {
    tableName: 'CONNECT',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
