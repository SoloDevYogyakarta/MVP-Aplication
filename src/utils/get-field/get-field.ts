import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
