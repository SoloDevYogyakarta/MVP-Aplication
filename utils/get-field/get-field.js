"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map