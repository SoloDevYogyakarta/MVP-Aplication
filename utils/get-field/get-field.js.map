{"version":3,"file":"get-field.js","sourceRoot":"","sources":["../../../src/utils/get-field/get-field.ts"],"names":[],"mappings":";;;AACA,6CAA4C;AAE5C,SAAgB,QAAQ,CAAC,QAAuB;IAC9C,IAAI;QACF,MAAM,GAAG,GAAG;YACV,MAAM;YACN,OAAO;YACP,cAAc;YACd,cAAc;YACd,gBAAgB;SACjB,CAAC;QACF,IAAI,GAAG,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE;YAC1B,OAAO,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC;SACnD;QACD,OAAO,IAAI,CAAC,KAAK,CAAC,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC,CAAC;KAC/D;IAAC,OAAO,GAAG,EAAE;QACZ,MAAM,IAAI,KAAK,CAAC,GAAG,CAAC,CAAC;KACtB;AACH,CAAC;AAhBD,4BAgBC"}