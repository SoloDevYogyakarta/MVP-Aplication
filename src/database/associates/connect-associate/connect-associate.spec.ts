import { ModelCtor } from 'sequelize';
import { ConnectInstance } from '../../../database/entities/commons/connect-entity/connect-entity';
import { getField } from '../../../utils/get-field/get-field';
import {
  connectAssociate,
  connectAttribute,
  connectFileInclude,
} from './connect-associate';

describe('connectAssociate', () => {
  let entity: ModelCtor<ConnectInstance>;

  beforeEach(() => {
    entity = connectAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('findOne with relationship', async () => {
      const { public_id } = getField('connect-entity');
      const findOne = await entity.findOne({
        where: { public_id },
        attributes: connectAttribute,
        include: connectFileInclude,
      });
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll({
        attributes: connectAttribute,
        include: connectFileInclude,
      });
    });
  } catch (err) {
    // empty
  }
});
