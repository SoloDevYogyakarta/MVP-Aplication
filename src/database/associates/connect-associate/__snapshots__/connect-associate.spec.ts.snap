// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectAssociate render correctly 1`] = `[Function]`;
